Build assets with Webpack

DFurnes opened the pull request:
#### What's this PR do?

This PR updates Phoenix to use the [shared Webpack config](https://github.com/DoSomething/webpack-config) and [ESLint config](https://github.com/DoSomething/eslint-config) that were extracted from Forge in DoSomething/forge#535. Because the Modernizr CLI & Webpack now handle building all of our front-end assets, this also removes Grunt as a build dependency.
#### How should this be manually tested?

If you want to manually test: pull down this branch, run `npm install` to get the new build dependencies, and then `npm start` to build & watch, or `npm run build` to build for production.

(The `ds grunt` alias that's used in our deploy script was also updated, although the command name is probably a bit suspect now!)
#### Any background context you want to provide?

References DoSomething/forge#535.

---

For review: @DoSomething/front-end

DFurnes committed:
Use NPM scripts rather than Grunt CLI.

DFurnes committed:
Use shared ESLint config & Sasslint CLI.

DFurnes committed:
Use Modernizr v3.

DFurnes committed:
Remove 404ed asset paths.

DFurnes committed:
Use shared Webpack config.

DFurnes committed:
ES6 modules must be imported via ES6 import.

Babel 6 now exposes the default ES6 export as the `[‘default’]` key on
the resulting object, so normal CommonJS imports no longer work. Let’s
just update these to ES6 imports to simplify things.

See: https://phabricator.babeljs.io/T2212

weerd commented on bin/ds:
```diff
@@ -447,9 +447,9 @@ function grunt {
       echo -e "\n\e[4mBuilding SCSS/JS with Grunt, and watching for changes...\e[0m"
       echo -e "\e[30;43m(Press Ctrl-C to stop.)\e[0m\n"
-      eval "/usr/bin/grunt"
+      eval "/usr/bin/npm start"
```
```
npm activate
```

```
npm commence
```

```
npm let'er rip
```

```
npm launch
```

```
npm initiate launch sequence
```

```
npm go
```

```
npm npmification
```

```
npm gotta catch em all
```

just tossing those out 😼

DFurnes replied:
![](https://media.giphy.com/media/M80pcQOnjh0YM/giphy.gif)

weerd commented on lib/themes/dosomething/paraneue_dosomething/scss/content/_finder.scss:
```diff
@@ -213,7 +213,7 @@
 
   .no-result {
-    background: $off-black asset-url("images/placeholder.png") center center;
+    background: $off-black center center;
```
why'd the default image get removed?

DFurnes replied:
It [doesn't exist](https://github.com/DoSomething/phoenix/tree/dev/lib/themes/dosomething/paraneue_dosomething/images)! Since Webpack loads assets as well (to potentially inline as Data URIs), it was failing because the path didn't resolve to anything.

weerd replied:
So it's fine to use the shorthand CSS property without defining an image path?

DFurnes replied:
I think so, although it doesn't really make any sense to set background positioning now. :confused:

weerd replied:
yar :\

angaither commented on lib/themes/dosomething/paraneue_dosomething/scss/content/_win-module.scss:
```diff
@@ -1,13 +1,11 @@
 .win-module {
   .win-module__progress {
     margin-bottom: $base-spacing;
-    background-image: asset-url("images/WinBackgroundS.svg");
```
rip win

DFurnes replied:
Oh, although this one actually _does_ exist. We just don't have a loader configured for SVGs. Is the "win module" still a thing?

angaither replied:
I don't think so if this is winning from a hot module. we still have the code for hot module, but we aren't using it anymoar
but there's a win state on campaigns... dunno which this is.

angaither commented:
@DFurnes can you update the [wiki](https://github.com/DoSomething/phoenix/wiki/Front-End-Development) does `ds grunt --watch` still work? 
we should probs change the name if it's a :suspect:

DFurnes commented:
@angaither I've left it as `ds grunt` and `ds grunt --watch` just so we don't trip anyone up. I think it could make sense to rename to something more tool-agnostic like `ds front-end` or `ds assets` maybe?

angaither commented:
@DFurnes as long as documentation is updated, it should be fine. esp if it doesn't make sense to leave it named that
my 2 :moneybag:

weerd commented:
yeah, rename it! `ds assets` works for me but might be too generic... `ds fender`?

DFurnes committed:
Remove unused shorthand properties.

DFurnes committed:
Rename "ds grunt" script to "ds assets".

Update the DS script for front-end assets now that we’re no longer
using Grunt behind the scenes.

DFurnes commented:
@angaither @weerd Updated! I left in compatibility with running `ds grunt` in case people have muscle-memory for that, but will update the docs once I merge! :smile:

weerd commented on bin/ds:
```diff
@@ -644,9 +640,9 @@ fi
 # grunt
 #----------------------------------------------------------------------
-if [[ $1 == "grunt" ]]
+if [[ $1 == "grunt" ]] || [[ $1 == "assets" ]]
```
:wink:

weerd commented on lib/themes/dosomething/paraneue_dosomething/scss/content/_finder.scss:
```diff
@@ -213,7 +213,7 @@
 
   .no-result {
-    background: $off-black asset-url("images/placeholder.png") center center;
+    background: $off-black;
```
Can we make this `background-color`? Would make it easy to customize adding to it if needed without having to override a bunch of the default properties the shorthand otherwise adds...

DFurnes replied:
Good call! Done!

DFurnes committed:
Use background-color rather than shorthand.

weerd commented:
:+1:

weerd commented:
:sheep: IT! :dancers:
